### ISSUE TITLE
esdoc fails with "Cannot read property 'indexOf' of undefined"

### ISSUE THREAD

**Issue by mattallty** [esdoc/esdoc#48]
esdoc fails on certain projects, but I can't see what is the cause.

```
/Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/Builder/ClassDocBuilder.js:79
            throw _iteratorError;
                  ^
TypeError: Cannot read property 'indexOf' of undefined
    at ClassDocBuilder._buildSignatureHTML (/Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:1052:26)
    at /Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:460:38
    at IceCap.loop (/Users/matt/dev/test/node_modules/esdoc/node_modules/ice-cap/out/src/IceCap.js:319:15)
    at ClassDocBuilder._buildSummaryDoc (/Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:457:11)
    at ClassDocBuilder._buildSummaryHTML (/Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/Builder/DocBuilder.js:416:29)
    at ClassDocBuilder._buildClassDoc (/Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/Builder/ClassDocBuilder.js:167:38)
    at ClassDocBuilder.exec (/Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/Builder/ClassDocBuilder.js:64:36)
    at publish (/Users/matt/dev/test/node_modules/esdoc/out/src/Publisher/publish.js:132:59)
    at Function.generate (/Users/matt/dev/test/node_modules/esdoc/out/src/ESDoc.js:199:7)
    at ESDocCLI.exec (/Users/matt/dev/test/node_modules/esdoc/out/src/ESDocCLI.js:88:28)
```

**Comment by h13i32maru** [comment]
@mattallty Hi! Thanks this issue :smile: 
Please comments the codes or repository url that raised this error.

**Comment by DarrenN** [comment]
I too am having this same issue. Unfortunately this is not public code I can post here.

**Comment by DarrenN** [comment]
Okay, I have narrowed it down, this block of code will crash ESDoc:

```javascript
export default class Alpha {
  /**
   * @param {Object} [donation]
   * @param {Object}
   */
  attachIdentity(donation={}) { }
};
```

The error is that there is a second `@param`, which should really be a `@return`. It took a while to find this problem. I think it would be better to provide a meaningful error message to the user in this case, or ignore the `@param` declaration for a param that doesn't exist.

Thoughts?

**Comment by h13i32maru** [comment]
@DarrenN Thanks!
> provide a meaningful error message

I will try to it, please wait a little.

**Comment by h13i32maru** [comment]
@mattallty @DarrenN I released ESDoc v0.2.3. The ESDoc show the following error log. Please try with your code.
<img width="371" alt="2015-08-29 17 37 53" src="https://cloud.githubusercontent.com/assets/985170/9561410/2c41d260-4e7f-11e5-9a27-8841eb3ee749.png">
